src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -56,10 +64,81 @@ llvm::Value* CodegenLLVMVisitor::codegen_indexed_name(const ast::IndexedName& no
     return create_gep(node.get_node_name(), index);
 }
 
+llvm::Value* CodegenLLVMVisitor::codegen_instance_var(const ast::CodegenInstanceVar& node) {
+    const auto& member_node = node.get_member_var();
+    const auto& instance_name = node.get_instance_var()->get_node_name();
+    const auto& member_name = member_node->get_node_name();
+
+    if (!instance_var_helper.is_an_instance_variable(member_name))
+        throw std::runtime_error("Error: " + member_name + " is not a member of the instance!");
+
+    // Load the instance struct given its name from the ValueSymbolTable.
+    llvm::Value* instance_ptr = builder.CreateLoad(lookup(instance_name));
+
+    // Create a GEP instruction to get a pointer to the member.
+    int member_index = instance_var_helper.get_variable_index(member_name);
+    llvm::Type* index_type = llvm::Type::getInt32Ty(*context);
+
+    std::vector<llvm::Value*> indices;
+    indices.push_back(llvm::ConstantInt::get(index_type, 0));
+    indices.push_back(llvm::ConstantInt::get(index_type, member_index));
+    llvm::Value* member_ptr = builder.CreateInBoundsGEP(instance_ptr, indices);
+
+    // Get the member AST node from the instance AST node, for which we proceed with the code
+    // generation. If the member is scalar, return the pointer to it straight away.
+    auto codegen_var_with_type = instance_var_helper.get_variable(member_name);
+    if (!codegen_var_with_type->get_is_pointer())
+        return member_ptr;
+
+    // Otherwise, the codegen variable is a pointer, and the member AST node must be an IndexedName.
+    auto member_var_name = std::dynamic_pointer_cast<ast::VarName>(member_node);
+    if (!member_var_name->get_name()->is_indexed_name())
+        throw std::runtime_error("Error: " + member_name + " is not an IndexedName!");
+
+    // Proceed to creating a GEP instruction to get the pointer to the member's element. While LLVM
+    // Helper set the indices to be Name nodes, a sanity check is added here. Note that this step
+    // can be avoided if using `get_array_index_or_length()`. However, it does not support indexing
+    // with Name/Expression at the moment. \todo: Reuse `get_array_index_or_length()` here.
+    auto member_indexed_name = std::dynamic_pointer_cast<ast::IndexedName>(
+        member_var_name->get_name());
+    if (!member_indexed_name->get_length()->is_name())
+        throw std::runtime_error("Error: " + member_name + " has a non-Name index!");
+
+    // Load the index variable that will be used to access the member's element. Since we index a
+    // pointer variable, we need to extend the 32-bit integer index variable to 64-bit.
+    llvm::Value* i32_index = builder.CreateLoad(
+        lookup(member_indexed_name->get_length()->get_node_name()));
+    llvm::Value* i64_index = builder.CreateSExt(i32_index, llvm::Type::getInt64Ty(*context));
```

Comment:
just curious - if index variable will be `size_t` type instead of integer, would that help?

Reply:
Do you mean update the type of the value in the node? In that case that does not make a difference. The sign extend is on IR level, independently of what is contained in the AST node. 

Why we cast to `i64` is an interesting question. From what I know, index bit width usually comes from the ABI information, data layout, etc. On CPU platforms this is usually 64 bit. So it is fine to use both 32 or 64 bit indices I think - I just reimplemented Clang's convention for indices, but `sext` can be dropped.

Reply:
Quoting LLVM Ref

Why index struct with `i32`?
> The specific type i32 is probably just a historical artifact, however it’s wide enough for all practical purposes, so there’s been no need to change it. It doesn’t necessarily imply i32 address arithmetic; it’s just an identifier which identifies a field in a struct. Requiring that all struct indices be the same reduces the range of possibilities for cases where two GEPs are effectively the same but have distinct operand types.

Otherwise we index with `i64`. Actually this just reminded me that our indices for arrays are `i32` that is not really conventional.

Reply:
partially that's because for the reason of memory saving (i32 is sufficient what we store).

Reply:
> Do you mean update the type of the value in the node? In that case that does not make a difference. The sign extend is on IR level, independently of what is contained in the AST node.
Yes. Ok.